drivers/serial/uart_stm32.c

```diff
@@ -32,6 +32,337 @@
+	if (IS_LPUART_INSTANCE(UartInstance)) {
+		return LL_LPUART_GetDataWidth(UartInstance);
+	} else {
+		return LL_USART_GetDataWidth(UartInstance);
+	}
+#else
+	return LL_USART_GetDataWidth(UartInstance);
+#endif	/* CONFIG_LPUART_1 */
+}
+
+static inline u32_t uart_stm32_cfg2ll_parity(enum uart_config_parity parity)
+{
+	switch (parity) {
+	case UART_CFG_PARITY_ODD:
+		return LL_USART_PARITY_ODD;
+	case UART_CFG_PARITY_EVEN:
+		return LL_USART_PARITY_EVEN;
+	case UART_CFG_PARITY_NONE:
+	default:
+		return LL_USART_PARITY_NONE;
+	}
+}
+
+static inline enum uart_config_parity uart_stm32_ll2cfg_parity(u32_t parity)
+{
+	switch (parity) {
+	case LL_USART_PARITY_ODD:
+		return UART_CFG_PARITY_ODD;
+	case LL_USART_PARITY_EVEN:
+		return UART_CFG_PARITY_EVEN;
+	case LL_USART_PARITY_NONE:
+	default:
+		return UART_CFG_PARITY_NONE;
+	}
+}
+
+static inline u32_t uart_stm32_cfg2ll_stopbits(enum uart_config_stop_bits sb)
+{
+	switch (sb) {
+/* Some MCU's don't support 0.5 stop bits */
+#ifdef LL_USART_STOPBITS_0_5
+	case UART_CFG_STOP_BITS_0_5:
+		return LL_USART_STOPBITS_0_5;
+#endif	/* LL_USART_STOPBITS_0_5 */
+	case UART_CFG_STOP_BITS_1:
+		return LL_USART_STOPBITS_1;
+/* Some MCU's don't support 1.5 stop bits */
+#ifdef LL_USART_STOPBITS_1_5
+	case UART_CFG_STOP_BITS_1_5:
+		return LL_USART_STOPBITS_1_5;
+#endif	/* LL_USART_STOPBITS_1_5 */
+	case UART_CFG_STOP_BITS_2:
+	default:
+		return LL_USART_STOPBITS_2;
+	}
+}
+
+static inline enum uart_config_stop_bits uart_stm32_ll2cfg_stopbits(u32_t sb)
+{
+	switch (sb) {
+/* Some MCU's don't support 0.5 stop bits */
+#ifdef LL_USART_STOPBITS_0_5
+	case LL_USART_STOPBITS_0_5:
+		return UART_CFG_STOP_BITS_0_5;
+#endif	/* LL_USART_STOPBITS_0_5 */
+	case LL_USART_STOPBITS_1:
+		return UART_CFG_STOP_BITS_1;
+/* Some MCU's don't support 1.5 stop bits */
+#ifdef LL_USART_STOPBITS_1_5
+	case LL_USART_STOPBITS_1_5:
+		return UART_CFG_STOP_BITS_1_5;
+#endif	/* LL_USART_STOPBITS_1_5 */
+	case LL_USART_STOPBITS_2:
+	default:
+		return UART_CFG_STOP_BITS_2;
+	}
+}
+
+static inline u32_t uart_stm32_cfg2ll_databits(enum uart_config_data_bits db)
+{
+	switch (db) {
+/* Some MCU's don't support 7B datawidth */
+#ifdef LL_USART_DATAWIDTH_7B
+	case UART_CFG_DATA_BITS_7:
+		return LL_USART_DATAWIDTH_7B;
+#endif	/* LL_USART_DATAWIDTH_7B */
+	case UART_CFG_DATA_BITS_8:
+	default:
+		return LL_USART_DATAWIDTH_8B;
+	}
+}
+
+static inline enum uart_config_data_bits uart_stm32_ll2cfg_databits(u32_t db)
+{
+	switch (db) {
+/* Some MCU's don't support 7B datawidth */
+#ifdef LL_USART_DATAWIDTH_7B
+	case LL_USART_DATAWIDTH_7B:
+		return UART_CFG_DATA_BITS_7;
+#endif	/* LL_USART_DATAWIDTH_7B */
+	case LL_USART_DATAWIDTH_8B:
+	default:
+		return UART_CFG_DATA_BITS_8;
+	}
+}
+
+static int uart_stm32_configure(struct device *dev,
+				const struct uart_config *cfg)
+{
+	struct uart_stm32_data *data = DEV_DATA(dev);
+	USART_TypeDef *UartInstance = UART_STRUCT(dev);
+	const u32_t parity = uart_stm32_cfg2ll_parity(cfg->parity);
```

Comment:
Can you check if we can save some code footprint by getting uart_stm32_cfg2ll_xxx functions to return and error code if requested valued is not supported and remove the checks below?

Reply:
I did a quick and dirty test with hello world on the nucleo_l4r5zi:
```
;; as is code
Memory region         Used Size  Region Size  %age Used
           FLASH:       13304 B         2 MB      0.63%
            SRAM:        4372 B       640 KB      0.67%
        IDT_LIST:         120 B         2 KB      5.86%

;; With cfg2ll functions returning error codes
Memory region         Used Size  Region Size  %age Used
           FLASH:       13320 B         2 MB      0.64%
            SRAM:        4372 B       640 KB      0.67%
        IDT_LIST:         120 B         2 KB      5.86%
```

I think the increase in codesize is due to error checks still needing to occur (to check the error of the cfg2ll functions), but now having more return codes out of those functions.